ambry-server/src/integration-test/java/com.github.ambry.server/ServerHardDeleteTest.java

```diff
@@ -273,7 +273,8 @@ public void endToEndTestHardDeletes() throws Exception {
     notificationSystem.awaitBlobDeletions(blobIdList.get(4).getID());
 
     time.sleep(TimeUnit.DAYS.toMillis(1));
-    ensureCleanupTokenCatchesUp(chosenPartition.getReplicaIds().get(0).getReplicaPath(), mockClusterMap, 198443);
+    // TODO: derive the token value automatically
```

Comment:
Just a note about this handcrafted value - the idea here was that we look for a verified value so that we guard against any unexpected change to this value due to code changes that are not supposed to change the token semantics. To that effect, deriving this automatically was intentionally avoided. This is a borrowed practice from the replication tests I believe.

If you have derived this value elsewhere, I would suggest taking a relook based on the above comment.